### DISCUSSION TITLE
Game pauses after scene transitions from previous scene but runs fine on its own

### DISCUSSION THREAD

**Discussion by abbylian** [dialogic-godot/dialogic#863] [Q&A]
Hiya, sorry for the poor title!

Scene C executes perfectly when I run it with 'Play the edited scene' but if I get to the scene via 'Play project' or a transition from Scene B, it comes to a total pause. All the other scenes in the prototype loads fine except Scene C. The difference between Scene A to B and Scene B to C is how it's done. All the scenes prior to Scene A and Scene B uses get_tree().change_scene while the transition from Scene B to Scene C is via Dialogic's Change Scene. Conversations that are coded into the script have unpause functions.

In case it's important: In Scene B, the player interacts has to interact with an NPC to get questioned a few times before getting sent to the next scene however they are also given the interaction to return to the current scene. Think of it as "Are you ready?" "A. Yes B. No". A brings up the questions, B lets the player roam around a bit more. There's no scores to keep track of so there's no condition for the scene to load either. I've included a close dialog in the timeline at the parent dialog(sorry, I'm not sure how to say this) too.

I followed the tutorial for the [code](https://imgur.com/a/cNBgxVx) to trigger it as a conversation with an NPC and this is the [timeline](https://imgur.com/a/sLnQjmQ).

**Comment by Jowan-Spooner** [comment]
Yeah this is a known problem with the code emilio used in his tutorial. The tree does not get unpaused before you change the scene. You can fix this by adding this to your code:

```
# where you add the dialog (after the NPC interaction)
    dialog.connect('dialogic_signal', self, '_on_dialogic_signal')

#...
# add the following function to the NPC script
func _on_dialogic_signal(argument):
    if argument == "unpause":
        get_tree().paused = false
```
Then add an Emit Signal event in the timeline before your Change Scene event and type `unpause` in the input field.

**Reply by abbylian** [reply]
Do you mean I should add `dialog.connect('dialogic_signal', self, '_on_dialogic_signal')` after 
`yield(dialog,'timeline_end')` or `add_child(dialog)` ? Sorry if the question is dumb!
